test/integration/stats_integration_test.cc

```diff
@@ -129,5 +164,33 @@ TEST_P(StatsIntegrationTest, WithTagSpecifierWithFixedValue) {
   EXPECT_EQ(live->tags()[0].value_, "xxx");
 }
 
+INSTANTIATE_TEST_SUITE_P(IpVersions, ClusterMemoryUtilization,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// TODO(cmluciano) Refactor once envoyproxy/envoy#5624 is solved
+// TODO(cmluciano) Add options to measure multiple workers & without stats
+TEST_P(ClusterMemoryUtilization, MemoryLargeClusterSizeWithStats) {
+  // Skip test if we cannot measure memory with TCMALLOC
+  if (!Stats::TestUtil::hasDeterministicMallocStats()) {
+    return;
+  }
+  const size_t start_mem = Memory::Stats::totalCurrentlyAllocated() / 1000;
+
+  auto IpVersions = testing::TestWithParam<Network::Address::IpVersion>::GetParam();
+
+  auto t1 = std::make_unique<BaseIntegrationTest>(IpVersions);
+  t1->initialize();
+  const size_t m1 = Memory::Stats::totalCurrentlyAllocated();
+  EXPECT_LT(start_mem, m1 / 1000);
+  EXPECT_LT(m1 / 1000, 3700); // actual value: 3664 as of Feb 26, 2019
+  t1.reset(nullptr);
+
+  const size_t m1001 = memoryConsumedWithClusters(1001, true);
+  EXPECT_LT(start_mem / 1000, m1001 / 1000);
+  size_t m_per_cluster = (m1001 - m1) / 1000;
+  EXPECT_LT(m_per_cluster, 58000); // actual value: 57936 as of Feb 26, 2019
```

Comment:
again, specify in comment by the libstdc++ and libc++ value. The reason is that when someone bloats Envoy and needs to fix this test, they need to know to update both values.